### DISCUSSION TITLE
Vue TypeScript type check

### DISCUSSION THREAD

**Discussion by masterT** [ElMassimo/vite_ruby#365] [Q&A]
Hi,

Since Vite does not perform type checking when using TypeScript, how would you perform type checks for Vue applications having **different custom properties**? 

- [custom properties Vue 2](https://v2.vuejs.org/v2/guide/typescript.html#Augmenting-Types-for-Use-with-Plugins)
- [custom properties Vue 3](https://vuejs.org/api/utility-types.html#componentcustomproperties)

Does anyone have an idea?

**Comment by ElMassimo** [comment]
Hi Simon!

You would typically use [`vue-tsc`](https://github.com/vuejs/language-tools/tree/master/packages/vue-tsc), running it in CI workflows or in commit hooks. [Example](https://github.com/ElMassimo/types_from_serializers/blob/main/playground/vanilla/package.json#L6).

**Reply by masterT** [reply]
Ok, this works fine if you only have one Vue application. 

In my case, I have 5 applications, each with its own Vue custom properties. 

I'm migrating from a custom webpack setup where all applications were built independently, so each has its own `tsconfig.json` file and its declaration file `d.ts` for the custom properties.

**Reply by ElMassimo** [reply]
You should be able to target specific directories with `vue-tsc`, though each app would need to be "self-contained".
